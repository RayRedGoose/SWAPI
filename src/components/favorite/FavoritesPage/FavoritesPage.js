import React from 'react'
import './FavoritesPage.scss';
import FavoriteCharacters from '../FavoriteCharacters/FavoriteCharacters'
import Header from '../../Header/Header'


const FavoritesPage = (props) => {
  return (
    <section className="favorite-page">
      <Header
        headerText={"My Favorite Characters"}
        user={props.user}
      />
      <FavoriteCharacters {...props} />
    </section>
  )
}


export default FavoritesPage;
