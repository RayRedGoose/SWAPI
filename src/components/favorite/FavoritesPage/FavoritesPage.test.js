import React from 'react'
import characterResults from '../../../mockData/fakeCharacters.js'
import { shallow } from 'enzyme'
import FavoritesPage from './FavoritesPage'

describe('FavoritesPage', () => {

  it('should render FavoritesPage with all data passed in correctly', () => {
    const wrapper = shallow(
      <FavoritesPage
        user={{
          name: 'Taylor Swift',
          quote: "I find your lack of faith disturbing",
          rank: 'intermediate'}}
        favorites={[characterResults[1]]}
        logOut={jest.fn()}
        removeFavorite={jest.fn()}
      />)

    expect(wrapper).toMatchSnapshot()
  })
})
