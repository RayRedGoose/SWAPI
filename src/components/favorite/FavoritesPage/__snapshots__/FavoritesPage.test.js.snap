// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage should render FavoritesPage with all data passed in correctly 1`] = `
<section
  className="favorite-page"
>
  <Header
    headerText="My Favorite Characters"
    user={
      Object {
        "name": "Taylor Swift",
        "quote": "I find your lack of faith disturbing",
        "rank": "intermediate",
      }
    }
  />
  <FavoriteCharacters
    favorites={
      Array [
        Object {
          "films": Array [
            "die hard",
          ],
          "isFavorite": true,
          "name": "Leia",
          "planet": "Alderaan",
          "population": 230000,
          "species": "human",
        },
      ]
    }
    removeFavorite={[MockFunction]}
  />
</section>
`;
